mesonbuild/backend/vs2015backend.py

```diff
@@ -19,5 +19,12 @@ class Vs2015Backend(Vs2010Backend):
     def __init__(self, build):
         super().__init__(build)
         self.name = 'vs2015'
-        self.platform_toolset = 'v140'
         self.vs_version = '2015'
+        comps = self.environment.coredata.compilers
+        if all(c.id == 'intel-cl' for c in comps.values()):
+            c = list(comps.values())[0]
+            if c.version.startswith('19'):
+                self.platform_toolset = 'Intel C++ Compiler 19.0'
+            # We don't have support for versions nolder than 2019 right now.
+        if self.platform_toolset is None:
```

Comment:
Unclear to me how this is going to handle combinations like 'CC=icl, CXX=cl'.  Are those unsupported?  Should that be mentioned in snippet?

Reply:
I've added an exception for that case, as well as one for older versions of ICL.

Reply:
I've also mentioned that in the docs.

Reply:
I've also mentioned that in the docs.